### PR TITLE
Doc/user: code color styles

### PR DESCRIPTION
**Pull request by joacoc**
### Motivation

Improve legibility by:
- Turning the `<code/>` font color similar to normal text but keeping the color when it is a link.
- Removing underlining in `<code/>` with links when there is no hover or active, like normal links.
- Setting the same underline color as the text
- Change `{{ note }}` background to match `{{ warning }}` and `{{ beta }}`

[Link](https://preview.materialize.com/materialize/17954/)

### Home
Screenshot 2023-03-06 at 13 15 57
Screenshot 2023-03-06 at 13 22 54

### Sources
Screenshot 2023-03-06 at 13 17 09
Screenshot 2023-03-06 at 13 21 59

### Notes
Screenshot 2023-03-06 at 13 18 01
Screenshot 2023-03-06 at 13 19 00

### Tips for reviewer

Color suggestions and improvements are super welcome!

I'm still deciding between background and code font color.

### Checklist

- [ ] This PR has adequate test coverage / QA involvement has been duly considered.
- [ ] This PR has an associated up-to-date [design doc](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/README.md), is a design doc ([template](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/00000000_template.md)), or is sufficiently small to not require a design.
  
- [ ] This PR evolves [an existing `$T ⇔ Proto$T` mapping](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/command-and-response-binary-encoding.md) (possibly in a backwards-incompatible way) and therefore is tagged with a `T-proto` label.
- [ ] If this PR will require changes to cloud orchestration, there is a companion cloud PR to account for those changes that is tagged with the release-blocker label ([example](https://github.com/MaterializeInc/cloud/pull/5021)).
  
- [ ] This PR includes the following [user-facing behavior changes](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/guide-changes.md#what-changes-require-a-release-note):
  -

### CODE REVIEW

**Comment by parkerhendo** [comment]
@joacoc I think I preferred this example you shared in slack where the code had a gray background.

I'm not sure if the monospace font + light gray border is enough to communicate that it's syntax. it's a bit hard to tell with the images, but I think we should reserve the purple color exclusively for links to. It'll help reduce visual noise and highlight the proper things.

This is the example I'm referring to (I prefer the one on the right)…
image

File: doc/user/assets/sass/_base.scss [hunk 1]
```diff
@@ -168,24 +176,64 @@ input[type="submit"] {
...
-a {
-    color: var(--sub);
+/**
+    The following section handles these cases:
+    1. Buttons with links
+    2. Links
+    3. Links with <code/>
+    4. Link in the index (<nav/>) with <code/>
+    5. Links inside notes
+    6. Links inside notes iwth <code/>
```
Comment (ruf-io) [inline doc/user/assets/sass/_base.scss:186]:
```suggestion
    6. Links inside notes with <code/>
```

File: doc/user/assets/sass/_content.scss [hunk 2]
```diff
@@ -977,11 +984,12 @@ body.dark .content {
...
     .note {
         a {
-            color: #4f0cff;
+            color: var(--link);
         }
 
         &::after {
-            border-color: var(--bg) var(--bg) var(--gray-mid) var(--note);
+            border-color: --var(--note-border);
+            // border-color: var(--bg) var(--bg) var(--gray-mid) var(--note);
```
Comment (ruf-io) [inline doc/user/assets/sass/_content.scss:992]:
```suggestion
            border-color: --var(--note-border);
```

**Review by ruf-io — APPROVED** [review]
Two tiny cleanups, otherwise LGTM!

**Pull request merged**